### PR TITLE
Fix transport test failures in Windows due to server initialization errors

### PR DESCRIPTION
**Pull request by sdkottegoda**
## Purpose
The server startup hangs intermittently in windows without a specific reason. The fix is to retry server startup for 3 attempts and fail if it still doesn't start. This fixes the test failures due to server initialization failure in windows.

### CODE REVIEW

File: integration/automation-extensions/src/main/java/org/wso2/esb/integration/common/extensions/carbonserver/CarbonServerManager.java [hunk 1]
```diff
@@ -148,7 +150,17 @@ public synchronized void startServerUsingCarbonHome(String carbonHome, Map<Strin
             waitTill(() -> !isRemotePortInUse("localhost", managementPort), 180, TimeUnit.SECONDS);
 
             if (!isRemotePortInUse("localhost", managementPort)) {
-                throw new RuntimeException("Server initialization failed");
+                if (retryAttempt < retryLimit){
+                    retryAttempt++;
+                    serverShutdown(portOffset, true);
+                    startServerUsingCarbonHome(carbonHome, commandMap);
```
Comment (Nirothipan) [inline integration/automation-extensions/src/main/java/org/wso2/esb/integration/common/extensions/carbonserver/CarbonServerManager.java]: Shall we add some info logs when we retry

File: integration/automation-extensions/src/main/java/org/wso2/esb/integration/common/extensions/carbonserver/CarbonServerManager.java [hunk 2]
```diff
@@ -148,7 +150,17 @@ public synchronized void startServerUsingCarbonHome(String carbonHome, Map<Strin
...
+                }
+                else {
+                    throw new RuntimeException("Server initialization failed");
```
Comment (Nirothipan) [inline integration/automation-extensions/src/main/java/org/wso2/esb/integration/common/extensions/carbonserver/CarbonServerManager.java]: do we need to set retryAttempt to zero here also ?

File: integration/automation-extensions/src/main/java/org/wso2/esb/integration/common/extensions/carbonserver/CarbonServerManager.java [hunk 3]
```diff
@@ -148,7 +150,17 @@ public synchronized void startServerUsingCarbonHome(String carbonHome, Map<Strin
...
+                }
+            }
+            else{
```
Comment (Nirothipan) [inline integration/automation-extensions/src/main/java/org/wso2/esb/integration/common/extensions/carbonserver/CarbonServerManager.java]: Shall we format this code

File: integration/automation-extensions/src/main/java/org/wso2/esb/integration/common/extensions/carbonserver/CarbonServerManager.java (hunk 1 above)
Comment (sdkottegoda) [reply integration/automation-extensions/src/main/java/org/wso2/esb/integration/common/extensions/carbonserver/CarbonServerManager.java]: done

File: integration/automation-extensions/src/main/java/org/wso2/esb/integration/common/extensions/carbonserver/CarbonServerManager.java (hunk 2 above)
Comment (sdkottegoda) [reply integration/automation-extensions/src/main/java/org/wso2/esb/integration/common/extensions/carbonserver/CarbonServerManager.java]: done

**Pull request merged**